pinot-common/src/main/java/org/apache/pinot/common/utils/TarCompressionUtils.java

```diff
@@ -69,6 +69,12 @@ public class TarCompressionUtils {
   private TarCompressionUtils() {
   }
 
+  /**
+   * This generic compressed tar file extension does not bind to a particular compressor. Decompression determines the
+   * appropriate compressor at run-time based on the file's magic number irrespective of the file extension.
+   * Compression uses the default compressor automatically if this generic extension is used.
+   */
+  public static final String TAR_COMPRESSED_FILE_EXTENSION = ".tar.compressed";
```

Comment:
mostly good. we are able to compress and decompress different segment format.

Curious, if we exposed and updated the value of _defaultCompressorName  (line 86), how can we make sure the `.tar.compressed` files can still be decompress by updated compressor?

Reply:
The unit tests should be inside Apache commons library. You can name the file extension to whatever you want, such as `.tar.deemoliu` and you'd still be able to decompress the segment. Decompression does not rely on the file extension to figure out the compressor to use for decompression.

Reply:
sounds good, so we using the first bytes to identify which decompressor to use. the apache library initializes the correct compressor for us.